### PR TITLE
Userland+Ports: Work towards building the bash port inside Serenity

### PR DESCRIPTION
**Pull request by BertalanD**
This PR fixes a couple of bugs in LibArchive, LibC and LibCore, and adds a missing feature to `date` in order to allow `bash`'s `configure` script to run inside SerenityOS.

After this PR is merged, only the following steps are needed to build bash:
- install the `bash`, `sed`, `patch`, `gawk`, `make`, `gcc`, `binutils` and `curl` ports
- symlink `/bin/sh` to a POSIX-compliant shell (bash, dash, oksh etc.)
- install the ports tree inside the VM (`ninja install-ports`)
- inside Serenity, copy the `/usr/Ports` directory to a place where `anon` has write access

Work on making this process less painful will take place in another PR. At the moment, a couple autoconf tests will crash and one will hang for 60 seconds (tracked in #12729), but the build will succeed regardless.

### CODE REVIEW

File: Userland/Utilities/date.cpp [hunk 1]
```diff
@@ -43,15 +45,21 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 0;
     }
 
-    // FIXME: this should be improved and will need to be cleaned up
-    // when additional output formats and formatting is supported
-    if (print_unix_date && print_iso_8601 && print_rfc_3339 && print_rfc_5322) {
-        warnln("date: multiple output formats specified");
+    if (print_unix_date + print_iso_8601 + print_rfc_3339 + print_rfc_5322 + !format_string.is_null() > 1) {
```
Comment (BertalanD) [inline Userland/Utilities/date.cpp:48]: This feels slightly wrong, but I don't know how to express more elegantly that at most 1 of these `bool`s can be true.

Comment (ADKaster) [reply Userland/Utilities/date.cpp:48]: the integer conversion should work. Using pure boolean logic would look pretty gnarly.
https://stackoverflow.com/a/33268481

File: Kernel/Process.cpp [hunk 2]
```diff
@@ -607,6 +610,9 @@ void Process::finalize()
     if (veil_state() == VeilState::Dropped)
         dbgln("\x1b[01;31mProcess '{}' exited with the veil left open\x1b[0m", name());
 
+    if (g_init_pid != 0 && pid() == g_init_pid)
```
Comment (BertalanD) [inline Kernel/Process.cpp:613]: We can't use a global `Optional` as that wants to call `__cxa_atexit`. PID 0 is always the colonel, so using 0 as a sentinel value here should be safe. Is there a better way to do this?

Comment (ADKaster) [reply Kernel/Process.cpp:613]: I would expect pid 0 to be init_stage2, but sure. You could use a NeverDestroyed<Optional<PID>> if you like. Probably best to have it store the Distinct numeric type.

**Review by linusg — APPROVED** [review]

**Pull request merged**